core/embed/rust/src/ui/model_mercury/component/status_screen.rs

```diff
@@ -1,13 +1,76 @@
-use crate::ui::{
-    component::{Component, Event, EventCtx, Timeout},
-    display::{Color, Icon},
-    geometry::{Alignment2D, Rect},
-    shape,
-    shape::Renderer,
+use crate::{
+    time::{Duration, Stopwatch},
+    ui::{
+        component::{Component, Event, EventCtx, Timeout},
+        constant::screen,
+        display::{Color, Icon},
+        geometry::{Alignment2D, Insets, Rect},
+        lerp::Lerp,
+        shape,
+        shape::Renderer,
+    },
 };
 
 use super::{theme, Swipe, SwipeDirection};
 
+const TIMEOUT_MS: u32 = 2000;
+
+#[derive(Default, Clone)]
+struct StatusAnimation {
+    pub timer: Stopwatch,
+}
+
+impl StatusAnimation {
+    pub fn is_active(&self) -> bool {
+        self.timer
+            .is_running_within(Duration::from_millis(TIMEOUT_MS))
+    }
+
+    pub fn eval(&self) -> f32 {
+        self.timer.elapsed().to_millis() as f32 / 1000.0
+    }
+
+    pub fn get_instruction_opacity(&self, t: f32) -> u8 {
+        let instruction_opacity = pareen::constant(0.0).seq_ease_in_out(
+            0.0,
+            easer::functions::Cubic,
+            0.42,
+            pareen::constant(1.0),
+        );
+        u8::lerp(0, 255, instruction_opacity.eval(t))
+    }
+
+    pub fn get_content_opacity(&self, t: f32) -> u8 {
+        let content_opacity = pareen::constant(0.0).seq_ease_in_out(
+            0.18,
+            easer::functions::Cubic,
+            0.2,
+            pareen::constant(1.0),
+        );
+
+        u8::lerp(0, 255, content_opacity.eval(t))
+    }
+
+    pub fn get_circle_scale(&self, t: f32) -> i16 {
+        let circle_scale = pareen::constant(0.0).seq_ease_out(
+            0.2,
+            easer::functions::Cubic,
+            0.4,
+            pareen::constant(1.0),
+        );
+
+        i16::lerp(170 / 2, 80 / 2, circle_scale.eval(t))
```

Comment:
well, would name these constants at least `170`, `80`. And not clear why divided by two. 
Not blocking the merge over this.

Reply:
https://github.com/trezor/trezor-firmware/pull/3807/commits/a65f355680f7a7404520b104253d5546e638ec13